package-scope.qmd

```diff
@@ -1,55 +1,43 @@
 
-## Epiverse-TRACE reasoning
+Epiverse packages may vary from these principles on a case by case basis. Packages that do must document the motivation in the package vignette.
 
-A benefit of including plotting functions is it enables package users to quickly inspect a feature of the package (e.g. model output) without having to write their own script or function. Despite this clear benefit of shipping plotting with a package there are several
-drawbacks, and here we outline the reasons why only dependency-free plotting (also commonly referred to as base R plotting) is included in Epiverse-TRACE R packages.
+## Reasoning
 
-The most frequently used plotting package in R is [{ggplot2}](https://ggplot2.tidyverse.org/index.html). This package implements the grammar of graphics and provides users with a extensive and flexible set of tools to build custom plots. These are the reasons it is used for plotting in package vignettes. However, in
-order for an R package to export plotting functions that use {ggplot2}, it must list this package
-as a [dependency](https://cran.r-project.org/doc/manuals/R-exts.html#Package-Dependencies). {ggplot2} is a "heavy" package, not so much from the size of the code base (as CRAN limits the size of packages to 5MB), but due to it's own dependencies.
+We include dependency free plotting in our packages (approach 1) to [minimize dependency bloat](https://epiverse-trace.github.io/blueprints/dependencies.html) and maximize maintainability. We must balance the benefits of dependencies with package development. 
 
-```{r, pkg-deps, eval=FALSE}
-tools::package_dependencies(packages = "ggplot2", recursive = FALSE)
+For example, we could include the most widely used plotting package, [{ggplot2}](https://ggplot2.tidyverse.org/index.html), as a dependency.
 
-tools::package_dependencies(packages = "ggplot2", recursive = TRUE)
-```
+However, {ggplot2} includes a fair amount of its own dependencies (`r length(tools::package_dependencies(packages = "ggplot2", recursive = TRUE)$ggplot2)` to be precise), risking dependency blaot. 
 
-The relatively large number of dependencies introduces a couple of issues:
+Within the wider Epiverse, we also [saw changes to {ggplot2}](https://github.com/reconhub/incidence/issues/119), breaking package implementations. This is always a risk of dependencies and goes against our [minimal dependency guideline](https://epiverse-trace.github.io/blueprints/dependencies.html).
 
-1. Breaking changes in, or deprecation of, {ggplot2} or one of it's dependencies may cause breaking changes in Epiverse-TRACE packages hampering sustainable package maintenance.
-2. If depended on, {ggplot2} would potentially install it's own dependencies on a user's device.
+Plots using {ggplot2} are immensely powerful. Package functions would either need to be highly specific or  complex to manage {ggplot2}'s functionality. We consider this out of scope, if we can provide code for {ggplot2} in vignettes directly.
```

Comment:
I would move this to the first paragraph of the Reasoning section, as I think it's the most important point around why we would/wouldn't exporting plotting functionality and whether this should use {ggplot2}.

I think we should mention that plotting functions exported by Epiverse packages should provide tangible value-add over just outputting data that can be used in calls to {ggplot2} functions. 

Pasting @Bisaloo's comment from #44 as I think it nicely makes this point:
> The question of using ggplot2 in demo vignettes or in the package core code IMO is more related to the fact that it's very difficult to wrap ggplot in a way that is simpler than pure ggplot but still keeps enough flexibility to be useful. Most of (all?) the examples that I have seen end up providing functions that are in no way simpler or more intuitive than direct ggplot use.
There are also considerations with tidy evaluation that are not strictly specific to ggplot wrapping, but more generally for using the tidyverse within the package.

Reply:
Moved it up! 👍 Thanks.